### PR TITLE
Typed Spans: Create RFC from current WIP proposal.

### PR DESCRIPTION
**Pull request by z1c0**
Since the "work in progress" folder is about to get removed from the repository (https://github.com/open-telemetry/opentelemetry-specification/pull/225), the proposal concerning **"typed spans"** will be moved from there into the RFC repo.

See original discussion: https://github.com/open-telemetry/opentelemetry-specification/pull/14

### CODE REVIEW

File: text/0000-typed-spans.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+- Destination Name
+- Channel Type
+- Channel Endpoint
+- Message Id
+- Correlation Id
+
+## Proposal
+* Add a field `CanonicalType` that contains the type of span
+* Define mandatory and optional attributes per span type
+* Provide an API that supports creating typed spans and ensures that at least all
```
[the remark this answers is not in the document]
Comment (AloisReitbauer) [reply text/0000-typed-spans.md]: Should we make language-specific RFCs? Not all languages support the same capabilities which makes writing specs for APIs a challenge. We can, however, link to the language spec files or should we rather use same pseudo code?

**Comment by AloisReitbauer** [comment]
Beyond the discussion above we should move these files over. This is just an administrative task rather than any substantial change to the spec. @tedsuo and @bogdandrutu  can you please approve.

**Comment by discostu105** [comment]
I believe this PR is a little bit ambiguous as it is. IMHO, it should be split into multiple standalone-pieces:

 1. Propose **canonical type** (with some concrete type-names)
 1. **Semantic conventions** per canonical type (these already exist, but maybe could use some refinements, better documenation, connection to concrete canonical types)
 1. Actual **typed span APIs** to help create spans that with proper attributes, that match semantic conventions for specific types.
 1. Fix **"component" field inconsistency**, because it partially overlaps with the "canonical type"
    - For HTTP, the definition is `component:http`, whereas for Database `component:<name-of-db`)
    - could also be part of 1.

**Comment by z1c0** [comment]
Closing this PR after some discussion. 

- We probably won't need a canonical type (we will check presence of attributes and apply a fixed resolution for multiple type spans, e.g. DB wins over HTTP).
- Semantic conventions are discussed separately in the Data Formats meeting.
- We should make a new OTEP/push for typed span APIs.

**Pull request closed without merging**